l10n_it_fatturapa_out_pec/models/fatturapa_attachment_out.py

```diff
@@ -0,0 +1,185 @@
+# -*- coding: utf-8 -*-
+# Author(s): Andrea Colangelo (andreacolangelo@openforce.it)
+# Copyright © 2018 Openforce Srls Unipersonale (www.openforce.it)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+import logging
+import re
+
+from lxml import etree
+
+from odoo import api, fields, models
+
+_logger = logging.getLogger(__name__)
+
+RESPONSE_MAIL_REGEX = '[A-Z]{2}[a-zA-Z0-9]{11,16}_[a-zA-Z0-9]{,5}_[A-Z]{2}_' \
+                      '[a-zA-Z0-9]{,3}'
+
+
+class FatturaPAAttachmentOut(models.Model):
+    _inherit = 'fatturapa.attachment.out'
+
+    state = fields.Selection([('ready', 'Ready to Send'),
+                              ('sent', 'Sent'),
+                              ('validated', 'Validated'),
+                              ('sender_error', 'Sender Error'),
+                              ('recipient_error', 'Recipient Error'),
+                              ('rejected', 'Rejected (PA)')],
+                             string='State',
+                             default='ready',)
+
+    last_sdi_response = fields.Text(
+        string='Last Response from Exchange System', default='No response yet')
+
+    @api.multi
+    def send_via_pec(self):
+
+        mail_message = self.env['mail.message'].create({
+            'model': self._name,
+            'res_id': self.id,
+            'subject': self.name,
+            'body': 'XML file for FatturaPA {} sent to Exchange System to the'
+                    ' email address {}.'
+            .format(
+                self.name,
+                self.env.user.company_id.email_exchange_system),
+            'attachment_ids': [(6, 0, self.ir_attachment_id.ids)],
+            'email_from': self.env.user.company_id.email_from_for_fatturaPA,
+            'mail_server_id': self.env.user.company_id.sdi_channel_id.
+            pec_server_id.id,
+        })
+
+        mail = self.env['mail.mail'].create({
+            'mail_message_id': mail_message.id,
+            'body_html': mail_message.body,
+            'email_to': self.env.user.company_id.email_exchange_system,
+        })
+
+        if mail:
+            res = mail.send()
+            if res:
+                self.state = 'sent'
+        # TODO Should we disable some mail.* config params before
+        # TODO sending? See: https://tinyurl.com/ybr45fxd
```

Comment:
@sergiocorato @warp10  questo l'avete verificato? Nel senso: se vengono impostati `mail.bounce.alias`, `mail.catchall.domain` e `mail.catchall.alias` (vedi https://www.agilebg.com/blog/blog-1/post/basic-email-configuration-for-odoo-7) l'invio via PEC funziona?

Reply:
@sergiocorato @warp10 Ho appena provato a inviare impostando `mail.bounce.alias`, `mail.catchall.domain` e `mail.catchall.alias` e ottengo
```
Mail delivery failed via SMTP server 'smtps.pec.aruba.it'.
SMTPRecipientsRefused: {'agilebg@pec.it': (553, '5.7.1 <elbaddy+1-fatturapa.attachment.out-6@gmail.com>: Sender address rejected: not owned by user l.battistini@pec.it')}
Traceback (most recent call last):
  File "/home/elbati/workspace/odoo/instances/demo10-community/parts/odoo/addons/mail/models/mail_mail.py", line 278, in send
    res = IrMailServer.send_email(msg, mail_server_id=mail.mail_server_id.id)
  File "/home/elbati/workspace/odoo/instances/demo10-community/parts/odoo/odoo/addons/base/ir/ir_mail_server.py", line 467, in send_email
    raise MailDeliveryException(_("Mail Delivery Failed"), msg)
MailDeliveryException: (u'Mail Delivery Failed', u"Mail delivery failed via SMTP server 'smtps.pec.aruba.it'.\nSMTPRecipientsRefused: {'agilebg@pec.it': (553, '5.7.1 <elbaddy+1-fatturapa.attachment.out-6@gmail.com>: Sender address rejected: not owned by user l.battistini@pec.it')}")
```

Reply:
Nello specifico i parametri sono
```
mail.catchall.alias | elbaddy
mail.bounce.alias | elbaddy
mail.catchall.domain | gmail.com
```